src/sentry/workflow_engine/processors/data_condition_group.py

```diff
@@ -127,7 +127,8 @@ def process_data_condition_group(
     logic_result, condition_results = evaluate_data_conditions(conditions_to_evaluate, logic_type)
 
     if (not logic_result and logic_type == DataConditionGroup.Type.ALL) or (
-        logic_result and logic_type == DataConditionGroup.Type.ANY
+        logic_result
+        and logic_type in (DataConditionGroup.Type.ANY, DataConditionGroup.Type.ANY_SHORT_CIRCUIT)
```

Comment:
🤔 

```python
is_short_circuit_all = not logic_result and logic_type == DataConditionGroup.Type.ALL
is_short_circuit_any = logic_result and logic_type in (DataConditionGroup.Type.ANY, DataConditionGroup.Type.ANY_SHORT_CIRCUIT)
        
if (is_short_circuit_all or is_short_circuit_any):
```
Just to help with readability, should we break apart the logic statements in the if?

Reply:
yes we can